website/static/js/nodesDelete.js

```diff
@@ -0,0 +1,275 @@
+            if (newValue[key].changed !== self.nodesOriginal[key].changed){
+                newValue[key].changed = true;
+                nodesDeleted++;
+            }
+            else {
+                newValue[key].changed = false;
+            }
+        }
+        self.nodesDeleted(nodesDeleted > 0 && nodesDeleted == Object.keys(self.nodesOriginal).length);
+        m.redraw(true);
+    });
+    //original node state on page load
+    self.hasChildren = ko.observable(false);
+    $('#nodesDelete').on('hidden.bs.modal', function () {
+        self.clear();
+    });
+
+    self.page = ko.observable(self.SELECT);
+
+    self.pageTitle = ko.computed(function() {
+        if (self.page() === self.WARNING &&  self.parentIsEmbargoed) {
+            return "This is a an embargo?";
+        }
+
+        return {
+            select: 'Delete Project',
+            confirm: 'Delete Project and Components'
+        }[self.page()];
+    });
+
+    self.message = ko.computed(function() {
+        if (self.page() === self.WARNING &&  self.parentIsEmbargoed) {
+            return "This is an embargo?"
+        }
+
+        if (self.page() === self.WARNING &&  self.isPreprint) {
+            return "You are attempting to delete a preprint."
+        }
+
+        if (self.page() === self.CONFIRM) {
+            return "The following project and components will be deleted"
+        }
+
+        return {
+            select: "It looks like your project has components within it. To delete this project, you must also delete all child components",
+            confirm: "The following project and components will be deleted."
+        }[self.page()];
+    });
+};
+
+/**
+ * get node tree for treebeard from API V1
+ */
+NodesDeleteViewModel.prototype.fetchNodeTree = function() {
+    var self = this;
+
+    return $.ajax({
+        url: self.treebeardUrl,
+        type: 'GET',
+        dataType: 'json'
+    }).done(function(response) {
+        self.nodesOriginal = getNodesOriginal(response[0], self.nodesOriginal);
+        var size = 0;
+        $.each(Object.keys(self.nodesOriginal), function(_, key) {
+            if (self.nodesOriginal.hasOwnProperty(key)) {
+                size++;
+            }
+        });
+        self.hasChildren(size > 1);
+        var nodesState = $.extend(true, {}, self.nodesOriginal);
+        var nodeParent = response[0].node.id;
+        self.nodesState(nodesState);
+    }).fail(function(xhr, status, error) {
+        $osf.growl('Error', 'Unable to retrieve project settings');
+        Raven.captureMessage('Could not GET project settings.', {
+            extra: {
+                url: self.treebeardUrl, status: status, error: error
+            }
+        });
+    });
+};
+
+NodesDeleteViewModel.prototype.selectProjects = function() {
+    this.page(this.SELECT);
+};
+
+NodesDeleteViewModel.prototype.confirmWarning =  function() {
+    var nodesState = ko.toJS(this.nodesState);
+    for (var node in nodesState) {
```

Comment:
This looks like a feature that was in the "change to public" that isn't really needed here. If they are using the delete, they will need a confirmation modal, whereas in the original they could try to change everything to public, but if everything was already public, then it wouldn't need a confirmation modal.

Reply:
The reason I copied and pasted from nodesPrivacy.js is that the feature specified in the ticket looks a lot like changing nodes privacy in that users have to manually choose and check all the project components shown in the project hierarchy to be able to proceed to next page for confirmation.

Reply:
I am not sure I understand. confirmWarning() changes the modal to a warning page where the titles of the project and its components are listed for user reference and asks for further confirmation. This is a specific feature request in the ticket.

Reply:
Gotcha. This one's good; I was thinking it was checking something else.